packages/presentation/lib/src/star_movie_app.dart

```diff
@@ -17,21 +19,29 @@ class _StarMovieAppState extends BlocScreenState<StatefulWidget, AppBloc> {
     return MaterialApp(
       title: 'Star Movie',
       theme: ThemeData.from(colorScheme: dark),
-      home: StreamBuilder(
+      home: StreamBuilder<BaseTile<AppData>>(
         stream: bloc.dataStream,
         builder: (context, snapshot) {
-          final baseTile = snapshot.data as BaseTile?;
+          final baseTile = snapshot.data;
           final tile = baseTile?.tile;
-          if (tile == null)
+          if (tile == null || baseTile == null)
             return Center(
               child: CircularProgressIndicator(),
             );
-          return Navigator(
-            onPopPage: (route, result) {
-              bloc.handleRemoveRouteSettings(route.settings);
-              return route.didPop(result);
-            },
-            pages: tile.pages.toList(),
+          return Scaffold(
+            bottomNavigationBar: baseTile.isLoading
+                ? null
+                : AppNavigationBar(
```

Comment:
can u please explain, how do u handle the cases when screen doesn't have bottom navigation? like splash screen for ex

Reply:
so, decided to try this:
class _AppBloc extends BlocImpl<BaseArguments, AppData> implements AppBloc {
  _AppBloc() : super(AppData.init());

  @override
  void init() {
    super.init();
    _update(isLoading: true);
    _initNavHandler();
  }
isLoading flag is on only on splash screen => bottom navBar is not showing. okay, I will think about alternative ways.

Reply:
not the best solution, because u can have the loading at any screen. u can check Katya's or Kolya's implementation, their variants seem prettier

Reply:
reworked

Reply:
seems strange

1) u have flag `isShowNavBar`. seems that when it is true, bottom navigation should be visible, otherwise - no. but in your case it works in the opposite way

<img width="303" alt="image" src="https://user-images.githubusercontent.com/64923802/191196666-2ae51eb2-0c46-4c23-9c5c-0b930751b1e8.png">

2) if I understood your code correctly, u hide navigation only at the first page and after any other action in navigator u show bottom navigation. this solution doesn't cover all cases - for example what if we will have a login screen without bottom navigation? recheck please Katya's pr, it covers this case

Reply:
Now I understand me mistake)